src/main/java/org/opensearch/knn/index/codec/nativeindex/remote/RemoteIndexBuildStrategy.java

```diff
@@ -127,22 +174,163 @@ private BlobStoreRepository getRepository() throws RepositoryMissingException {
+     * @param vectorDataType            Data type of the vector (FLOAT, BYTE, BINARY)
+     * @param knnVectorValuesSupplier   Supplier for {@link KNNVectorValues}
      * @throws IOException
      * @throws InterruptedException
      */
-    private void writeToRepository(
-        String fieldName,
-        Supplier<KNNVectorValues<?>> knnVectorValuesSupplier,
+    @VisibleForTesting
+    void writeToRepository(
+        String blobName,
         int totalLiveDocs,
-        SegmentWriteState segmentWriteState
+        VectorDataType vectorDataType,
+        Supplier<KNNVectorValues<?>> knnVectorValuesSupplier
     ) throws IOException, InterruptedException {
-        throw new NotImplementedException();
+        // Get the blob container based on blobName and the repo base path. This is where the blobs will be written to.
+        BlobPath path = getRepository().basePath().add(indexSettings.getUUID() + VECTORS_PATH);
+        BlobContainer blobContainer = getRepository().blobStore().blobContainer(path);
+
+        KNNVectorValues<?> knnVectorValues = knnVectorValuesSupplier.get();
+        initializeVectorValues(knnVectorValues);
+        long vectorBlobLength = (long) knnVectorValues.bytesPerVector() * totalLiveDocs;
+
+        if (blobContainer instanceof AsyncMultiStreamBlobContainer) {
+            // First initiate vectors upload
+            log.debug("Repository {} Supports Parallel Blob Upload", getRepository());
+            // WriteContext is the main entry point into asyncBlobUpload. It stores all of our upload configurations, analogous to
+            // BuildIndexParams
+            WriteContext writeContext = new WriteContext.Builder().fileName(blobName + VECTOR_BLOB_FILE_EXTENSION)
+                .streamContextSupplier((partSize) -> getStreamContext(partSize, vectorBlobLength, knnVectorValuesSupplier, vectorDataType))
+                .fileSize(vectorBlobLength)
+                .failIfAlreadyExists(true)
+                .writePriority(WritePriority.NORMAL)
+                // TODO: Checksum implementations -- It is difficult to calculate a checksum on the knnVectorValues as
+                // there is no underlying file upon which we can create the checksum. We should be able to create a
+                // checksum still by iterating through once, however this will be an expensive operation.
+                .uploadFinalizer((bool) -> {})
+                .doRemoteDataIntegrityCheck(false)
+                .expectedChecksum(null)
+                .build();
+
+            AtomicReference<Exception> exception = new AtomicReference<>();
+            final CountDownLatch latch = new CountDownLatch(1);
+            ((AsyncMultiStreamBlobContainer) blobContainer).asyncBlobUpload(
+                writeContext,
+                new LatchedActionListener<>(new ActionListener<>() {
+                    @Override
+                    public void onResponse(Void unused) {
+                        log.debug(
```

Comment:
Im thinking this can be info as well - I dont think this will happen that frequently and will be good to know.